### PR TITLE
Island: Implement login using flask-security

### PR DESCRIPTION
**Pull request by cakekoa**
# What does this PR do?

Fixes part of #2157.

Add any further explanations here.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/app.py [hunk 1]
```diff
@@ -109,7 +110,15 @@ class CustomConfirmRegisterForm(ConfirmRegisterForm):
             "Email", default="dummy@dummy.com", validators=[validate_no_user_exists_already]
         )
 
-    app.security = Security(app, user_datastore, confirm_register_form=CustomConfirmRegisterForm)
+    login_manager = LoginManager()
+    login_manager.init_app(app)
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/app.py:114]: Why?

Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/app.py:114]: Do we not need this if we want to use flask-login? Or is that done by the LgoinForm?

Comment (VakarisZ) [reply monkey/monkey_island/cc/app.py:114]: We don't want to use `flask-login`, we want to use `flask-security-too`

File: monkey/tests/unit_tests/monkey_island/cc/resources/auth/test_login.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        ),
+    )
+
+    response = make_login_request(TEST_REQUEST)
+
+    assert response.status_code == 400
+    mock_authentication_service.unlock_repository_encryptor.assert_not_called()
+
+
+def test_authentication_error(make_login_request, mock_authentication_service):
```
Comment (ilija-lazoroski) [inline monkey/tests/unit_tests/monkey_island/cc/resources/auth/test_login.py:67]: Rename authentication to login

File: monkey/tests/unit_tests/monkey_island/cc/resources/auth/test_register.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 
 @pytest.fixture
-def make_auth_request(flask_client):
+def make_login_request(flask_client):
```
Comment (ilija-lazoroski) [inline monkey/tests/unit_tests/monkey_island/cc/resources/auth/test_register.py:14]: Rename login to register

**Review by ilija-lazoroski — APPROVED** [review]
Minor renaming and merge! GJ

**Pull request merged**